fastjet/clustersequence.go

```diff
@@ -360,11 +370,30 @@ func (cs *ClusterSequence) runN3Dumb() error {
 					cs.jetScaleForAlgorithm(ijet),
 					cs.jetScaleForAlgorithm(jjet),
 				)
-				y := jetscale * Distance(ijet, jjet) * cs.invR2
-				if y < ymin {
-					ymin = y
-					ii = i
-					jj = j
+				switch cs.alg {
+				case EeGenKtAlgorithm:
+					den := 1 - math.Cos(cs.r)
+					if cs.r > math.Pi {
+						den = 3 + math.Cos(cs.r)
+					}
+					y := math.MaxFloat64
+					if den != 0 {
+						y = jetscale * (1 - fmom.CosTheta(&ijet.PxPyPzE, &jjet.PxPyPzE)) / den
+					}
+					if y < ymin {
+						ymin = y
+						ii = i
+						jj = j
+					}
+					break
+				default:
+					y := jetscale * Distance(ijet, jjet) * cs.invR2
+					if y < ymin {
+						ymin = y
+						ii = i
+						jj = j
+					}
```

Comment:
the whole if-block is repeated for each of the 2 cases.

couldn't we declare the `y` variable before the `switch`, fill it with the correct value inside the `case` clauses, and move the `if y < ymin { ... }` block after the `switch` ?